### PR TITLE
Stop creating ApplicationRecord on model generation

### LINKED ISSUE #29407: application_record generation does not honor autoload paths
### Steps to reproduce

add a new directory to autoload paths (i.e. foo_model)
move application_record to this directory
create a new migration
a new copy of application_record will appear in models directory

### Expected behavior

If the constant ApplicationRecord can be found using the normal auto load mechanism then do not regenerate a new blank copy.

### Actual behavior
System seems to thave the path app/models hard coded as the location of application-record

### System configuration
5.0 +
any ruby version

### PR DESCRIPTION
**Pull request by lugray**
Fixes #29407.

### Summary
When generating models, we created ApplicationRecord in the default location if no file existed there.  That was annoying for people who moved it to somewhere else in the autoload path.  At this point, the vast majority of apps should have either run the upgrade script or generated a model since upgrading.  For those that haven't the error message after generating a new model should be helpful:
```
NameError: uninitialized constant ApplicationRecord
```

### Other Information
To ease friction in that case, this also adds a generator for ApplicationRecord.

### CODE REVIEW

File: activerecord/lib/rails/generators/active_record/application_record/application_record_generator.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require "rails/generators/active_record"
```
Comment (rafaelfranca) [inline activerecord/lib/rails/generators/active_record/application_record/application_record_generator.rb]: Missing frozen string literal comment

**Review by rafaelfranca — CHANGES_REQUESTED** [review]
This change will need a CHANGELOG entry

**Comment by lugray** [comment]
Updated.

**Review by rafaelfranca — APPROVED** [review]

**Comment by eileencodes** [comment]
Thanks @lugray! And also thank you for writing an excellent commit message 😸

**Pull request merged**